### ISSUE TITLE
Feature idea [CodeFix provider]: all fields, methods, and properties must be wrapped into appropriately named `#region`s

### ISSUE THREAD

**Issue by Aaronontheweb** [Aaronontheweb/EvilRoslynAnalyzers#12]
Per Mark's suggestion on Twitter:

> [Enforce] regions, but specifically by member type and visibility? Like “Constructors”, “Private Fields”, “Internal Methods” etc.
>
> Bonus points for preventing use of plurals when there’s only one thing in a region.

**Comment by ardalis** [comment]
Nah, plurals all the time, and every single one is required in every .cs file regardless of whether the region block is empty.

**Comment by Aaronontheweb** [comment]
> Nah, plurals all the time, and every single one is required in every .cs file regardless of whether the region block is empty.

That might make this a tad harder to implement but it's a worthy goal
